### PR TITLE
build: add actions for CI and publishing

### PR DESCRIPTION
**Pull request by xballoy**
Add GitHub actions to:

- build library on PR and when updating main
- publish library on npm

I also cleaned up the unused dependencies and bump the existing one.

It will look like this:

Screen Shot 2022-10-11 at 3 54 50 PM

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Test
+        run: yarn test
+
+      - name: Bump the package version
+        run: |
+          git config user.name 'XXX XXX'
+          git config user.email 'xxx@xxx.xxx'
```
Comment (xballoy) [inline .github/workflows/publish.yml]: @mewhit what Git configuration do you want to use here? It's used when you publish a version, npm publish will bump the `version` in the `package.json` file and the the push will push it to the repo.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          npm version ${{ github.event.inputs.version }}
+
+      - name: Build
+        run: yarn build
+
+      - name: Publish
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (xballoy) [inline .github/workflows/publish.yml:50]: @mewhit you need to generate a npm token and save it in your GitHub Secrets (see https://michaelzanggl.com/articles/github-actions-cd-setup/#npm-auth-token)

**Comment by xballoy** [comment]
@mewhit I think you need to allow my Action to run the 1st time (because it's my 1st contribution to the repo).

File: .github/workflows/publish.yml (hunk 1 above)
Comment (mewhit) [reply .github/workflows/publish.yml]: username: MeWhit.
email: mikewhittom27@gmail.com

**Pull request merged**